{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"provenance":[]},"kernelspec":{"name":"python3","display_name":"Python 3"},"language_info":{"name":"python"}},"cells":[{"cell_type":"code","execution_count":1,"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"1xMGgJBQYF0R","outputId":"42a6d270-e7be-4310-c2ad-fc00a348ccdb","executionInfo":{"status":"ok","timestamp":1708189899345,"user_tz":300,"elapsed":24880,"user":{"displayName":"Runtian Miao","userId":"10135715695019588434"}}},"outputs":[{"output_type":"stream","name":"stdout","text":["Python 3.10.12\n","Collecting git+https://github.com/bmschmidt/quadfeather\n","  Cloning https://github.com/bmschmidt/quadfeather to /tmp/pip-req-build-8ho8vzlu\n","  Running command git clone --filter=blob:none --quiet https://github.com/bmschmidt/quadfeather /tmp/pip-req-build-8ho8vzlu\n","  Resolved https://github.com/bmschmidt/quadfeather to commit 6336d068025fa38b40a0e63ecd525ab716fe8819\n","  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n","Requirement already satisfied: pyarrow in /usr/local/lib/python3.10/dist-packages (from quadfeather==1.1.2) (10.0.1)\n","Requirement already satisfied: numpy in /usr/local/lib/python3.10/dist-packages (from quadfeather==1.1.2) (1.25.2)\n","Requirement already satisfied: pandas in /usr/local/lib/python3.10/dist-packages (from quadfeather==1.1.2) (1.5.3)\n","Requirement already satisfied: python-dateutil>=2.8.1 in /usr/local/lib/python3.10/dist-packages (from pandas->quadfeather==1.1.2) (2.8.2)\n","Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.10/dist-packages (from pandas->quadfeather==1.1.2) (2023.4)\n","Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.10/dist-packages (from python-dateutil>=2.8.1->pandas->quadfeather==1.1.2) (1.16.0)\n","Building wheels for collected packages: quadfeather\n","  Building wheel for quadfeather (setup.py) ... \u001b[?25l\u001b[?25hdone\n","  Created wheel for quadfeather: filename=quadfeather-1.1.2-py3-none-any.whl size=12020 sha256=22897cf39548d7c7b1b9dfe760a1e1c0fdbc6ab6031e1c2777ee629d4e20b8f9\n","  Stored in directory: /tmp/pip-ephem-wheel-cache-lawjv0g_/wheels/cc/5f/13/e5ad5627ae7b56298f3f5f563a05feb509d010e9aeccf50222\n","Successfully built quadfeather\n","Installing collected packages: quadfeather\n","Successfully installed quadfeather-1.1.2\n"]}],"source":["!python3 -V # requires Python 3.9.x or 3.10.x\n","!python3 -m pip install git+https://github.com/bmschmidt/quadfeather\n","!quadfeather-test-data 1_000_000\n","!quadfeather --files tmp.csv --tile_size 50_000 --destination tiles"]},{"cell_type":"code","source":["!npm i\n","!npm run dev"],"metadata":{"id":"Kn-41671YdF1","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1708190746749,"user_tz":300,"elapsed":1462,"user":{"displayName":"Runtian Miao","userId":"10135715695019588434"}},"outputId":"60c67c04-69af-4796-956b-374c336ebd9c"},"execution_count":4,"outputs":[{"output_type":"stream","name":"stdout","text":["\u001b[?25l[\u001b[90m..................\u001b[0m] | rollbackFailedOptional: \u001b[34;40mverb\u001b[0m \u001b[35mnpm-session\u001b[0m 8d5513834a7b08a\u001b[0m\u001b[K\r[\u001b[7m       \u001b[27m\u001b[90m...........\u001b[0m] | postinstall: \u001b[7msill\u001b[0m \u001b[35minstall\u001b[0m executeActions\u001b[0m\u001b[K\r\r\u001b[K\u001b[?25h\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[30;43mWARN\u001b[0m \u001b[0m\u001b[35msaveError\u001b[0m ENOENT: no such file or directory, open '/content/package.json'\n","\u001b[0m\u001b[?25l[\u001b[7m        \u001b[27m\u001b[90m..........\u001b[0m] / prepare:content: \u001b[30;43mWARN\u001b[0m \u001b[35msaveError\u001b[0m ENOENT: no such file or\u001b[0m\u001b[K\r\r\u001b[K\u001b[?25h\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[30;43mWARN\u001b[0m \u001b[0m\u001b[35menoent\u001b[0m ENOENT: no such file or directory, open '/content/package.json'\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[30;43mWARN\u001b[0m\u001b[35m\u001b[0m content No description\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[30;43mWARN\u001b[0m\u001b[35m\u001b[0m content No repository field.\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[30;43mWARN\u001b[0m\u001b[35m\u001b[0m content No README data\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[30;43mWARN\u001b[0m\u001b[35m\u001b[0m content No license field.\n","\u001b[0m\n","\u001b[K\u001b[?25hup to date in 0.258s\n","found \u001b[92m0\u001b[0m vulnerabilities\n","\n","\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35mcode\u001b[0m ENOENT\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35msyscall\u001b[0m open\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35mpath\u001b[0m /content/package.json\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35merrno\u001b[0m -2\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35menoent\u001b[0m ENOENT: no such file or directory, open '/content/package.json'\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35menoent\u001b[0m This is related to npm not being able to find a file.\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m \u001b[0m\u001b[35menoent\u001b[0m \n","\u001b[0m\n","\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m\u001b[35m\u001b[0m A complete log of this run can be found in:\n","\u001b[0m\u001b[37;40mnpm\u001b[0m \u001b[0m\u001b[31;40mERR!\u001b[0m\u001b[35m\u001b[0m     /root/.npm/_logs/2024-02-17T17_25_46_538Z-debug.log\n","\u001b[0m"]}]},{"cell_type":"code","source":["import subprocess\n","\n","def run_npm_commands():\n","    # Change directory to your project folder if necessary\n","    project_dir = '/path/to/your/project'\n","    try:\n","        # Installing npm dependencies\n","        print(\"Installing npm dependencies...\")\n","        subprocess.run([\"npm\", \"install\"], check=True, cwd=project_dir)\n","        print(\"Dependencies installed successfully.\")\n","\n","        # Running the 'dev' script from package.json\n","        print(\"Starting development server...\")\n","        subprocess.run([\"npm\", \"run\", \"dev\"], check=True, cwd=project_dir)\n","        print(\"Development server started.\")\n","    except subprocess.CalledProcessError as e:\n","        print(f\"Command failed with error: {e}\")\n","\n","# Execute the function\n","run_npm_commands()\n"],"metadata":{"colab":{"base_uri":"https://localhost:8080/","height":385},"id":"HKJ9KSfEIENU","executionInfo":{"status":"error","timestamp":1708191141343,"user_tz":300,"elapsed":196,"user":{"displayName":"Runtian Miao","userId":"10135715695019588434"}},"outputId":"670f24f1-560d-4631-91be-7ab1cb4eeed9"},"execution_count":5,"outputs":[{"output_type":"stream","name":"stdout","text":["Installing npm dependencies...\n"]},{"output_type":"error","ename":"FileNotFoundError","evalue":"[Errno 2] No such file or directory: '/path/to/your/project'","traceback":["\u001b[0;31m---------------------------------------------------------------------------\u001b[0m","\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)","\u001b[0;32m<ipython-input-5-3377da8eeec5>\u001b[0m in \u001b[0;36m<cell line: 20>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     18\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     19\u001b[0m \u001b[0;31m# Execute the function\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 20\u001b[0;31m \u001b[0mrun_npm_commands\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m","\u001b[0;32m<ipython-input-5-3377da8eeec5>\u001b[0m in \u001b[0;36mrun_npm_commands\u001b[0;34m()\u001b[0m\n\u001b[1;32m      7\u001b[0m         \u001b[0;31m# Installing npm dependencies\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      8\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Installing npm dependencies...\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 9\u001b[0;31m         \u001b[0msubprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrun\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"npm\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"install\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcheck\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcwd\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mproject_dir\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     10\u001b[0m         \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Dependencies installed successfully.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/lib/python3.10/subprocess.py\u001b[0m in \u001b[0;36mrun\u001b[0;34m(input, capture_output, timeout, check, *popenargs, **kwargs)\u001b[0m\n\u001b[1;32m    501\u001b[0m         \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'stderr'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mPIPE\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    502\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 503\u001b[0;31m     \u001b[0;32mwith\u001b[0m \u001b[0mPopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0mpopenargs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mprocess\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    504\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    505\u001b[0m             \u001b[0mstdout\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstderr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mprocess\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcommunicate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minput\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/lib/python3.10/subprocess.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, args, bufsize, executable, stdin, stdout, stderr, preexec_fn, close_fds, shell, cwd, env, universal_newlines, startupinfo, creationflags, restore_signals, start_new_session, pass_fds, user, group, extra_groups, encoding, errors, text, umask, pipesize)\u001b[0m\n\u001b[1;32m    969\u001b[0m                             encoding=encoding, errors=errors)\n\u001b[1;32m    970\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 971\u001b[0;31m             self._execute_child(args, executable, preexec_fn, close_fds,\n\u001b[0m\u001b[1;32m    972\u001b[0m                                 \u001b[0mpass_fds\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcwd\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0menv\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    973\u001b[0m                                 \u001b[0mstartupinfo\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcreationflags\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mshell\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;32m/usr/lib/python3.10/subprocess.py\u001b[0m in \u001b[0;36m_execute_child\u001b[0;34m(self, args, executable, preexec_fn, close_fds, pass_fds, cwd, env, startupinfo, creationflags, shell, p2cread, p2cwrite, c2pread, c2pwrite, errread, errwrite, restore_signals, gid, gids, uid, umask, start_new_session)\u001b[0m\n\u001b[1;32m   1861\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0merrno_num\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1862\u001b[0m                         \u001b[0merr_msg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mos\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstrerror\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno_num\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1863\u001b[0;31m                     \u001b[0;32mraise\u001b[0m \u001b[0mchild_exception_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merrno_num\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr_msg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0merr_filename\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1864\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mchild_exception_type\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merr_msg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1865\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n","\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '/path/to/your/project'"]}]},{"cell_type":"code","source":["#!mkdir deepscatter\n","#!cd deepscatter # if you're not already there\n","!quadfeather --files /content/fiction.csv --tile_size 50000 --destination tiles"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"HdVITX7CYs9Y","outputId":"923ab82a-49d9-4fa0-cc58-6527feb3e633","executionInfo":{"status":"ok","timestamp":1708190053022,"user_tz":300,"elapsed":627,"user":{"displayName":"Runtian Miao","userId":"10135715695019588434"}}},"execution_count":3,"outputs":[{"output_type":"stream","name":"stdout","text":["Traceback (most recent call last):\n","  File \"/usr/local/bin/quadfeather\", line 8, in <module>\n","    sys.exit(main())\n","  File \"/usr/local/lib/python3.10/dist-packages/quadfeather/tiler.py\", line 221, in main\n","    schema, schema_safe = determine_schema(args)\n","  File \"/usr/local/lib/python3.10/dist-packages/quadfeather/tiler.py\", line 101, in determine_schema\n","    vals = csv.open_csv(args.files[0],\n","  File \"pyarrow/_csv.pyx\", line 1267, in pyarrow._csv.open_csv\n","  File \"pyarrow/_csv.pyx\", line 1077, in pyarrow._csv._get_reader\n","  File \"pyarrow/io.pxi\", line 1781, in pyarrow.lib.get_input_stream\n","  File \"pyarrow/io.pxi\", line 1752, in pyarrow.lib.get_native_file\n","  File \"pyarrow/io.pxi\", line 995, in pyarrow.lib.OSFile.__cinit__\n","  File \"pyarrow/io.pxi\", line 1005, in pyarrow.lib.OSFile._open_readable\n","  File \"pyarrow/error.pxi\", line 144, in pyarrow.lib.pyarrow_internal_check_status\n","  File \"pyarrow/error.pxi\", line 113, in pyarrow.lib.check_status\n","FileNotFoundError: [Errno 2] Failed to open local file '/content/fiction.csv'. Detail: [errno 2] No such file or directory\n"]}]},{"cell_type":"code","source":["!npm run dev"],"metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"mCdCsQ8QtbW0","outputId":"603caf05-3f65-44e6-b294-920b749ed634"},"execution_count":null,"outputs":[{"output_type":"stream","name":"stdout","text":["\n","> deepscatter@2.15.2 dev /content\n","> vite --port 3344 --host\n","\n","\n","\u001b[1;1H\u001b[0J\n","  \u001b[32m\u001b[1mVITE\u001b[22m v4.1.4\u001b[39m  \u001b[2mready in \u001b[0m\u001b[1m296\u001b[22m\u001b[2m\u001b[0m ms\u001b[22m\n","\n","  \u001b[32m➜\u001b[39m  \u001b[1mLocal\u001b[22m:   \u001b[36mhttp://localhost:\u001b[1m3344\u001b[22m/\u001b[39m\n","  \u001b[32m➜\u001b[39m  \u001b[1mNetwork\u001b[22m: \u001b[36mhttp://172.28.0.12:\u001b[1m3344\u001b[22m/\u001b[39m\n","\u001b[2m\u001b[32m  ➜\u001b[39m\u001b[22m\u001b[2m  press \u001b[22m\u001b[1mh\u001b[22m\u001b[2m to show help\u001b[22m\n"]}]}]}